src/hooks/useToast.tsx

```diff
@@ -0,0 +1,41 @@
+import { create } from "zustand";
+
+interface ToastItem {
+  id: number;
+  label: string;
+  isVisible: boolean;
+}
+
+interface ToastState {
+  toasts: ToastItem[];
+  showToast: (label: string) => void;
+  removeToast: (id: number) => void;
+}
+
+export const useToast = create<ToastState>((set) => ({
+  toasts: [],
+  showToast: (label) => {
+    const id = Date.now();
+    set((state) => ({
+      toasts: [...state.toasts, { id, label, isVisible: true }],
+    }));
+
+    setTimeout(() => {
+      set((state) => ({
+        toasts: state.toasts.map((toast) =>
+          toast.id === id ? { ...toast, isVisible: false } : toast,
+        ),
+      }));
+
+      setTimeout(() => {
+        set((state) => ({
+          toasts: state.toasts.filter((toast) => toast.id !== id),
+        }));
+      }, 500);
+    }, 1000);
+  },
```

Comment:
제 생각이지만 .. 중첩 `setTimeout`은 `async/await`으로 개선이 가능해보여요. 아래처럼 작성하면 구조가 간결화되어서 가독성이 좋아지지 않을까 합니당. Just 제안입니다!!

```suggestion
  showToast: async (label) => {
    const id = Date.now();
    set((state) => ({
      toasts: [...state.toasts, { id, label, isVisible: true }],
    }));
    await new Promise((r) => setTimeout(r, 1000));
    set((state) => ({
      toasts: state.toasts.map((t) =>
        t.id === id ? { ...t, isVisible: false } : t
      ),
    }));
    await new Promise((r) => setTimeout(r, 500));
    set((state) => ({
      toasts: state.toasts.filter((t) => t.id !== id),
    }));
  };
```

Reply:
말씀해주신 부분들 반영하여 코드 수정하도록 하겠습니다!!😊